src/python/pants/backend/java/lint/google_java_format/rules_integration_test.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+import pytest
+
+from pants.backend.java.compile.javac import rules as javac_rules
+from pants.backend.java.lint import java_fmt
+from pants.backend.java.lint.google_java_format import rules as gdf_fmt_rules
+from pants.backend.java.lint.google_java_format import skip_field
+from pants.backend.java.lint.google_java_format.rules import (
+    GoogleJavaFormatFieldSet,
+    GoogleJavaFormatRequest,
+)
+from pants.backend.java.target_types import JavaSourcesGeneratorTarget, JavaSourceTarget
+from pants.backend.java.target_types import rules as target_types_rules
+from pants.build_graph.address import Address
+from pants.core.goals.fmt import FmtResult
+from pants.core.goals.lint import LintResult, LintResults
+from pants.core.util_rules import config_files, source_files
+from pants.core.util_rules.external_tool import rules as external_tool_rules
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.fs import CreateDigest, Digest, FileContent
+from pants.engine.rules import QueryRule
+from pants.engine.target import Target
+from pants.jvm import classpath
+from pants.jvm.jdk_rules import rules as java_util_rules
+from pants.jvm.resolve.coursier_fetch import rules as coursier_fetch_rules
+from pants.jvm.resolve.coursier_setup import rules as coursier_setup_rules
+from pants.jvm.util_rules import rules as util_rules
+from pants.testutil.rule_runner import PYTHON_BOOTSTRAP_ENV, RuleRunner
+
+NAMED_RESOLVE_OPTIONS = '--jvm-resolves={"test": "coursier_resolve.lockfile"}'
+DEFAULT_RESOLVE_OPTION = "--jvm-default-resolve=test"
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    rule_runner = RuleRunner(
+        rules=[
+            *config_files.rules(),
+            *classpath.rules(),
+            *coursier_fetch_rules(),
+            *coursier_setup_rules(),
+            *external_tool_rules(),
+            *source_files.rules(),
+            *javac_rules(),
```

Comment:
Shouldn't need these.

Reply:
Only  `*external_tool_rules()` and `*source_files.rules()` could be removed without rule graph errors. Also added `jdk_rules` explicitly.